src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -42,40 +41,88 @@ internal ILGeneratorImpl(MethodBuilder methodBuilder, int size)
         public override LocalBuilder DeclareLocal(Type localType, bool pinned) => throw new NotImplementedException();
         public override Label DefineLabel() => throw new NotImplementedException();
 
-        public override void Emit(OpCode opcode)
+        private static int GetStackChangeFor(StackBehaviour stackBehaviour) =>
```

Comment:
If we make the below API public, we don't need this `GetStackChangeFor` method and calculating the `StackBehaviourPush`, `StackBehaviourPop` diff with  `GetStackChangeFor(opCode.StackBehaviourPush) + GetStackChangeFor(opCode.StackBehaviourPop)` https://github.com/dotnet/runtime/blob/715293eeeddf76593679fba7ab8cfd451980225b/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/Opcode.cs#L48-L49

Reply:
As per offline discussion created [API proposal](https://github.com/dotnet/runtime/issues/93419) for making this public, updated the PR accordingly